modules/launcher/src/main/java/org/mule/module/launcher/util/JdbcDriverManagerInitializer.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.module.launcher.util;
+
+import static java.lang.Boolean.TRUE;
+import static java.lang.ClassLoader.getSystemClassLoader;
+import static java.sql.DriverManager.getDrivers;
+import static org.mule.api.config.MuleProperties.SYSTEM_PROPERTY_PREFIX;
+import org.mule.util.PropertyChecker;
+
+import java.sql.DriverManager;
+
+/**
+ * Initializes JDBC {@link DriverManager} class in order to prevent {@link ClassLoader} leaks.
+ *
+ * @since 3.8.2
+ */
+public class JdbcDriverManagerInitializer
+{
+    public static final String JDBC_DRIVER_MANAGER_INITIALIZER = SYSTEM_PROPERTY_PREFIX + "jdbcDriverManagerInitializer";
+
+    /**
+     * The first access to {@link DriverManager} will trigger the loading of
+     * all {@link java.sql.Driver}s in the the current class loader.
+     * Triggering the loading here has fewer side-effects.
+     */
+    protected static PropertyChecker JDBC_DRIVER_MANAGER_INITIALIZER_PROPERTY = new PropertyChecker(JDBC_DRIVER_MANAGER_INITIALIZER, TRUE.toString());
+
+
+    /**
+     * Initializes JDBC {@link DriverManager} in order to avoid memory leaks when disposing applications.
+     * Uses the system {@link ClassLoader} to load the {@link Class}.
+     */
+    public void initialize()
+    {
+        final ClassLoader loader = Thread.currentThread().getContextClassLoader();
+
+        try
+        {
+            // Use the system classloader as the victim for all this
+            // ClassLoader pinning we're about to do.
+            Thread.currentThread().setContextClassLoader(getSystemClassLoader());
+
+            /*
+             * First call to this loads all drivers in the current class
+             * loader
+             */
+            if (JDBC_DRIVER_MANAGER_INITIALIZER_PROPERTY.isEnabled())
+            {
+                getDrivers();
```

Comment:
I remember some code was added to the application classloader or something like that in order to release resources on application undeployment caused by drivers that were registered but never unregistered. Is not this code going to cause that same issue we fixed?